Initial standalone RabbitMQ implementation

bdero opened the pull request:
This PR implements a simple Ansible role to setup a non-clustered standalone RabbitMQ. This was adapted from the ansible-examples repo.

This playbook is already deployed on rabbitmq-dev.opencraft.hosting, rabbitmq-stage.opencraft.hosting, and rabbitmq.opencraft.hosting.

**Testing instructions:**

1. Clone [`deployment-secret-services`](https://github.com/open-craft/deployment-secrets-services) and checkout the [`bdero/rabbitmq`](https://github.com/open-craft/deployment-secrets-services/pull/13) branch.
1. Setup `deployment-secret-services` using the [readme instructions](https://github.com/open-craft/deployment-secrets-services#preparatory-steps).
1. In the **hosts** file, comment out the `rabbitmq.opencraft.hosting` and `rabbitmq-stage.opencraft.hosting` lines.
1. Run the playbook over **rabbitmq-dev.opencraft.hosting**. Replace `$SHARED_MYSQL_IDENT` with the location of the shared DB key (use the "ssh key for databases instances" entry in keepass):
   ```
   ansible-playbook -vvv deploy/deploy-all.yml -u ubuntu --private-key $SHARED_MYSQL_IDENT -i hosts -l rabbitmq --vault-password-file .vault-pass
   ```

bdero committed:
Cleaned up from ansible examples

haikuginger commented on tasks/main.yml:
```diff
@@ -0,0 +1,48 @@
+
+- name: add rabbitmq trusted key
+  apt_key:
+    url: http://www.rabbitmq.com/rabbitmq-release-signing-key.asc
```
HTTPS here too, please.

bdero replied:
Done.

haikuginger commented on tasks/main.yml:
```diff
@@ -0,0 +1,48 @@
+
+- name: add rabbitmq repository
+  apt_repository:
+    repo: 'deb http://www.rabbitmq.com/debian/ testing main'
```
HTTPS, please.

bdero replied:
Done.

haikuginger reviewed (commented):
This code looks pretty good in general, with a couple caveats. We just want to make sure to be using HTTPS wherever possible.

bdero commented:
@haikuginger Ready for another review - thanks! I also added instructions to the PR description to aid with testing setup.

haikuginger commented:
Realizing - if we're provisioning new users using Ansible, do we necessarily need to create the admin user? From what I'm seeing, it doesn't seem like any admin username/password variables are passed [in here.](https://github.com/open-craft/opencraft/pull/152/files#diff-80ecfdb45c52624924d13fcf0dfc8077R2)

If we don't need the user to create other new users, and we're not planning on snooping on any communications with it, then the secure option is to not create a superuser-like user in the first place.

bdero commented:
@haikuginger Nope, we don't _need_ the admin user - it's just there to allow for navigating through the web view and diagnosing problems like that. Of course that's not necessary, since `rabbitmqctl` can just be used, so probably we can just remove it in favor of not having it open. Thanks.

bdero commented:
@haikuginger We now need the `admin` user, as I refactored the RabbitMQ OCIM PR to use the HTTP RabbitMQ API instead of configuring via playbook.

haikuginger commented:
I noticed that you're using a Xenial image to run this, which doesn't appear to come with Python 2. When starting up a new instance to run this on, I came across what appeared to be a relevant error. Please implement something like this to prevent this going forward:

http://stackoverflow.com/questions/32429259/ansible-fails-with-bin-sh-1-usr-bin-python-not-found

haikuginger commented:
@bdero, this failed when running on a new instance due to LetsEncrypt not reaching it at the expected domain name (which is fine and expected). Anyway, per your instructions, I went to run it on the existing dev instance and ended up stuck at this failure:

```Using module file /Users/haikuginger/deployment-secrets-services/ansible/lib/python2.7/site-packages/ansible/modules/core/packaging/os/apt.py
<149.202.168.174> ESTABLISH SSH CONNECTION FOR USER: ubuntu
<149.202.168.174> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="db_key.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ubuntu -o ConnectTimeout=10 -o ControlPath=/Users/haikuginger/.ansible/cp/ansible-ssh-%h-%p-%r 149.202.168.174 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1480701615.35-204024462661558 `" && echo ansible-tmp-1480701615.35-204024462661558="` echo $HOME/.ansible/tmp/ansible-tmp-1480701615.35-204024462661558 `" ) && sleep 0'"'"''
<149.202.168.174> PUT /var/folders/ck/thcptfm1789cgcl6dnkcfj480000gn/T/tmpoBJSAj TO /home/ubuntu/.ansible/tmp/ansible-tmp-1480701615.35-204024462661558/apt.py
<149.202.168.174> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="db_key.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ubuntu -o ConnectTimeout=10 -o ControlPath=/Users/haikuginger/.ansible/cp/ansible-ssh-%h-%p-%r '[149.202.168.174]'
<149.202.168.174> ESTABLISH SSH CONNECTION FOR USER: ubuntu
<149.202.168.174> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="db_key.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ubuntu -o ConnectTimeout=10 -o ControlPath=/Users/haikuginger/.ansible/cp/ansible-ssh-%h-%p-%r 149.202.168.174 '/bin/sh -c '"'"'chmod u+x /home/ubuntu/.ansible/tmp/ansible-tmp-1480701615.35-204024462661558/ /home/ubuntu/.ansible/tmp/ansible-tmp-1480701615.35-204024462661558/apt.py && sleep 0'"'"''
<149.202.168.174> ESTABLISH SSH CONNECTION FOR USER: ubuntu
<149.202.168.174> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="db_key.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ubuntu -o ConnectTimeout=10 -o ControlPath=/Users/haikuginger/.ansible/cp/ansible-ssh-%h-%p-%r -tt 149.202.168.174 '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-ntrpxvqcwiabbylcsuppkbxdfodrgsus; /usr/bin/python /home/ubuntu/.ansible/tmp/ansible-tmp-1480701615.35-204024462661558/apt.py; rm -rf "/home/ubuntu/.ansible/tmp/ansible-tmp-1480701615.35-204024462661558/" > /dev/null 2>&1'"'"'"'"'"'"'"'"' && sleep 0'"'"''
fatal: [rabbitmq-dev.opencraft.hosting]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_name": "apt"
    }, 
    "module_stderr": "Shared connection to 149.202.168.174 closed.\r\n", 
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_LmREVl/ansible_module_apt.py\", line 909, in <module>\r\n    main()\r\n  File \"/tmp/ansible_LmREVl/ansible_module_apt.py\", line 892, in main\r\n    if updated_cache and not retvals['changed']:\r\nKeyError: 'changed'\r\n", 
    "msg": "MODULE FAILURE"
}
```

Could you look into that?

bdero commented:
@haikuginger 

> I noticed that you're using a Xenial image to run this, which doesn't appear to come with Python 2. When starting up a new instance to run this on, I came across what appeared to be a relevant error.

 Ah, this was because I didn't add rabbitmq to the pattern for the "Install Python 2" play in `deploy-all.yml`. I added it in and now it'll install python2:

```
ansible-playbook -vvv deploy/deploy-all.yml -u ubuntu --private-key ~/.ssh/id_rsa_opencraft_mysql -i hosts -l rabbitmq --vault-password-file .vault-pass --list-hosts
Using /home/bdero/projects/opencraft/deployment-secrets-services/ansible.cfg as config file
4 plays in deploy/deploy-all.yml

playbook: deploy/deploy-all.yml

  play #1 (load-balancer:rabbitmq): Install Python 2	TAGS: []
    pattern: [u'load-balancer:rabbitmq']
    hosts (1):
      rabbitmq-dev.opencraft.hosting
```
As for the apt failure, I had the wrong RabbitMQ version in the defaults, so I've updated it and I'm redeploying to make sure all is well with the rest of the play.

haikuginger commented:
:+1:

- [x] I tested this: by running the role across the existing rabbitmq-dev server
- [x] I read through the code
- [x] ~~I checked for accessibility issues~~ N/A
- [x] ~~Includes documentation~~ N/A
